src/hooks/useSession.tsx

```diff
@@ -0,0 +1,211 @@
+  const { apiClient } = useHttpClient();
+  const userConfiguration = useMemo(
+    () =>
+      userSessionInfo
+        ? (JSON.parse(userSessionInfo) as UserConfiguration)
+        : configurationPlaceholder,
+    [userSessionInfo],
+  );
+
+  const fetchNewSession = useCallback(async () => {
+    const fetchedSessionInfo = await fetchAllUserSessionInfo(apiClient);
+    setUserSessionInfo(JSON.stringify(fetchedSessionInfo));
+    setIsLoaded(true);
+  }, [apiClient, setUserSessionInfo]);
+
+  useEffect(() => {
+    if (isStorageLoaded && !!authResult?.accessToken) {
+      const lastWeek = new Date();
+      lastWeek.setDate(lastWeek.getDate() - 7);
+      if (
+        !userSessionInfo ||
+        new Date(userConfiguration.sessionTime) < lastWeek
+      ) {
+        __DEV__ &&
+          console.info(
+            'No session detected or session expired, loading new session',
+          );
+        fetchNewSession();
+      } else {
+        setIsLoaded(true);
+      }
+    }
+  }, [
+    apiClient,
+    authResult?.accessToken,
+    fetchNewSession,
+    isStorageLoaded,
+    setUserSessionInfo,
+    userConfiguration.sessionTime,
+    userSessionInfo,
+  ]);
+
+  const clearSession = useCallback(() => {
+    setIsLoaded(false);
+    clearItem();
+  }, [clearItem]);
+
+  return (
+    <SessionContext.Provider
+      value={{
+        userConfiguration: userConfiguration,
+        isLoaded:
+          isLoaded && userConfiguration?.user.id !== Session.placeholder,
+        clearSession,
+      }}
+    >
+      {children}
+    </SessionContext.Provider>
+  );
+};
+
+export const useSession = () => useContext(SessionContext);
+
+const filterNonLRAccounts = (accounts?: Account[]) =>
+  accounts?.filter((a) => a.products?.indexOf('LR') > -1) || [];
+
+const fetchAllUserSessionInfo = async (
+  apiClient: APIClient<RestAPIEndpoints>,
+): Promise<UserConfiguration> => {
+  const user = await apiClient.request('GET /v1/user', {});
+  const accounts = await apiClient.request('GET /v1/accounts', {});
+  const accountsWithProduct = filterNonLRAccounts(accounts.data.accounts);
+  const configurations = await Promise.all(
+    accountsWithProduct.map(async (account) => {
+      const accountHeader = { 'LifeOmic-Account': account.id };
+      const useMeResponse = await apiClient.request(
+        'GET /v1/fhir/dstu3/$me',
+        {},
+        { headers: accountHeader },
+      );
+      const subjects = patientsToSubjects(useMeResponse.data.entry);
+      const subjectsWithProjects = await Promise.all(
+        subjects.map(async (s) => {
+          const projectResponse = await apiClient.request(
+            'GET /v1/projects',
+            {
+              id: s.projectId,
+            },
+            { headers: accountHeader },
+          );
+          const project = projectResponse.data.items[0];
+          const appConfig = await apiClient.request(
+            'GET /v1/life-research/projects/:projectId/app-config',
```

Comment:
We could fetch these at the same time by assuming the `s.projectId` is valid.

Reply:
Great point!